### PR TITLE
chore: sctpd can be packaged via bazel into a deb file

### PR DESCRIPTION
**Pull request by nstng**
Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

## Summary

This change allows to build an sctpd debian package with Bazel as in `lte/gateway/release/build-magma.sh`. The package is currently not used (uploaded etc) but build regularly with the Bazel workflow in bazel.yml (`bazel build ...` builds everything).

This first step introduces Bazel rules for building packages. It is planned to use these mechanics also for the magma debian package and the third party packages that are distributed in the magma artifactory.

There are certain differences between a script created package [1] and a package created with Bazel [2]:
* the bazel file has no timestamp in it's file name - in order to create reproducible builds
* Bazel sets all timestamps to `2000-01-01` - in order to create reproducible builds
* Bazel does not package changelog.gz - this seems to happen automatically with `fpm` - there was no relevant content in the make created changelog.gz

[1] `$ dpkg -c magma-sctpd_1.8.0-1653290604-_amd64.deb`
```
drwxr-xr-x 0/0               0 2022-05-23 09:27 ./
drwxr-xr-x 0/0               0 2022-05-23 09:27 ./etc/
drwxr-xr-x 0/0               0 2022-05-23 09:27 ./etc/systemd/
drwxr-xr-x 0/0               0 2022-05-23 09:27 ./etc/systemd/system/
-rw-r--r-- 0/0            1224 2021-06-18 01:45 ./etc/systemd/system/sctpd.service
drwxr-xr-x 0/0               0 2022-05-23 09:27 ./usr/
drwxr-xr-x 0/0               0 2022-05-23 09:27 ./usr/local/
drwxr-xr-x 0/0               0 2022-05-23 09:27 ./usr/local/sbin/
-rwxr-xr-x 0/0        20385568 2022-05-23 09:19 ./usr/local/sbin/sctpd
drwxr-xr-x 0/0               0 2022-05-23 09:27 ./usr/local/share/
drwxr-xr-x 0/0               0 2022-05-23 09:27 ./usr/local/share/sctpd/
-rw------- 0/0              18 2022-05-23 09:27 ./usr/local/share/sctpd/version
drwxr-xr-x 0/0               0 2022-05-23 09:27 ./usr/share/
drwxr-xr-x 0/0               0 2022-05-23 09:27 ./usr/share/doc/
drwxr-xr-x 0/0               0 2022-05-23 09:27 ./usr/share/doc/magma-sctpd/
-rw-r--r-- 0/0             153 2022-05-23 09:27 ./usr/share/doc/magma-sctpd/changelog.gz
```

[2] `$ dpkg -c bazel-bin/lte/gateway/release/magma-sctpd_1.8.0_amd64.deb`
```
drwxr-xr-x 0/0               0 2000-01-01 00:00 ./etc/
drwxr-xr-x 0/0               0 2000-01-01 00:00 ./etc/systemd/
drwxr-xr-x 0/0               0 2000-01-01 00:00 ./etc/systemd/system/
-rw-r--r-- 0/0            1224 2000-01-01 00:00 ./etc/systemd/system/sctpd.service
drwxr-xr-x 0/0               0 2000-01-01 00:00 ./usr/
drwxr-xr-x 0/0               0 2000-01-01 00:00 ./usr/local/
drwxr-xr-x 0/0               0 2000-01-01 00:00 ./usr/local/sbin/
-rwxr-xr-x 0/0       119582032 2000-01-01 00:00 ./usr/local/sbin/sctpd
drwxr-xr-x 0/0               0 2000-01-01 00:00 ./usr/local/share/
drwxr-xr-x 0/0               0 2000-01-01 00:00 ./usr/local/share/sctpd/
-rw-r--r-- 0/0               6 2000-01-01 00:00 ./usr/local/share/sctpd/version
```

## Test Plan

* Build: `bazel build lte/gateway/release:sctpd_deb_pkg`
* Creates files:
```
$ ll bazel-bin/lte/gateway/release/
-r-xr-xr-x 1 vscode vscode  715 May 29 16:37 magma-sctpd_1.8.0_amd64.changes*
-r-xr-xr-x 1 vscode vscode 115M May 29 16:37 magma-sctpd_1.8.0_amd64.deb*
-r-xr-xr-x 1 vscode vscode 115M May 29 16:37 magma-sctpd_1.8.0_amd64.tar*
-r-xr-xr-x 1 vscode vscode  324 May 29 16:37 sctpd_content.manifest*
lrwxrwxrwx 1 vscode vscode   97 May 29 16:37 sctpd_deb_pkg.deb -> /tmp/bazel/execroot/__main__/bazel-out/k8-dbg/bin/lte/gateway/release/magma-sctpd_1.8.0_amd64.deb*
-r-xr-xr-x 1 vscode vscode    6 May 29 16:37 version*
```
* Install on (e.g.) magma vm: apt install `./magma-sctpd_1.8.0_amd64.deb`
* see that files are setup as in [2] - start sctpd service and check logging

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by LKreutzer** [comment]
Do we know why the bazel-built `./usr/local/sbin/sctpd` is almost six times as large?

File: lte/gateway/release/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ARCH = "amd64"
+
+FILE_NAME = "{name}_{ver}_{arch}".format(
+    name = SCTPD_PKGNAME,
+    arch = ARCH,
+    ver = VERSION,
+)
+
+genrule(
+    name = "gen_scptd_version",
```
Comment (LKreutzer) [inline lte/gateway/release/BUILD.bazel]: Can we fix the spelling here and in the other targets, it should be "sctpd" instead of "scptd"?

**Review by LKreutzer — CHANGES_REQUESTED** [review]

File: lte/gateway/release/BUILD.bazel (hunk 1 above)
Comment (nstng) [reply lte/gateway/release/BUILD.bazel]: o0 wups, sure - good catch

**Review by LKreutzer — APPROVED** [review]
Tested locally, lgtm (modulo typos)!

File: lte/gateway/release/BUILD.bazel (hunk 1 above)
Comment (nstng) [reply lte/gateway/release/BUILD.bazel]: fixed

**Comment by nstng** [comment]
> Do we know why the bazel-built `./usr/local/sbin/sctpd` is almost six times as large?

The following libraries are linked dynamically in a make build (and statically in a bazel build)
```
libgflags.so.2.2 => /lib/x86_64-linux-gnu/libgflags.so.2.2
libglog.so.0 => /lib/x86_64-linux-gnu/libglog.so.0
libgpr.so.6 => /usr/local/lib/libgpr.so.6
libgrpc++.so.1 => /usr/local/lib/libgrpc++.so.1
libgrpc.so.6 => /usr/local/lib/libgrpc.so.6
liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5
libprotobuf.so.17 => /lib/x86_64-linux-gnu/libprotobuf.so.17
librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1
libsentry.so => /usr/local/lib/libsentry.so
libunwind.so.8 => /lib/x86_64-linux-gnu/libunwind.so.8
libyaml-cpp.so.0.6 => /lib/x86_64-linux-gnu/libyaml-cpp.so.0.6
```
Together they make up ca 15mb - and they again have dynamically linked libraries. -> The bazel build is more hermetically and thus has less environment dependencies.

But, your question made have a more detailed look - sctpd for bazel deb builds need addition flags -> set back to draft for now. :+1:

**Review by LKreutzer — APPROVED** [review]

**Pull request merged**